cdc/sink/codec/avro.go

```diff
@@ -379,19 +392,35 @@ func columnToAvroNativeData(col *model.Column) (interface{}, string, error) {
 
 	switch col.Type {
 	case mysql.TypeDate, mysql.TypeDatetime, mysql.TypeNewDate, mysql.TypeTimestamp:
-		str := col.Value.(string)
-		t, err := time.Parse(types.DateFormat, str)
+		// Refer to `unflatten` in cdc/entry/codec.go for why this piece of code is like this.
 		const fullType = "long." + timestampMillis
+		str := col.Value.(string)
+
+		if (col.Type == mysql.TypeDate && str == zeroDateStr) ||
+			(col.Type != mysql.TypeDate && str == zeroTimeStr) {
+
+			return time.Time{}, string(fullType), nil
```

Comment:
```suggestion
			return nil, string(fullType), nil
```

Why not nil?

Reply:
`nil` does not have the same type as `time.Time{}`, which will cause problem in the caller of this function.